### PR TITLE
feat(security-email): add support for security email

### PR DESCRIPTION
**Pull request by mwasilew2**
## Description of the change

related to: https://app.clickup.com/t/8693fq8dx

Adds to the terraform provider support for editing security email.

Note: the graphql api doesn't currently support deleting security email and terraform test sdk runs delete after each test

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (non-breaking change that adds documentation)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development
- [ ] Examples for new resources and data sources have been added
- [ ] Default values have been documented in the description (e.g., "Dummy: (Boolean) Blah blah. Defaults to `false`.)
- [ ] If the action fails that checks the documentation: Run `go generate` to make sure the docs are up to date

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] Pull Request is no longer marked as "draft"
- [ ] Reviewers have been assigned
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Comment by Clickup-user** [comment]
Task linked: [CU-8693fq8dx add support for `security_email` to terraform provider](https://app.clickup.com/t/8693fq8dx)

**Comment by mwasilew2** [comment]
similar to: https://github.com/spacelift-io/terraform-provider-spacelift/pull/495

File: spacelift/resource_security_email.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	if err := i.(*internal.Client).Mutate(ctx, "AccountUpdateSecurityEmail", &mutation, variables); err != nil {
+		return diag.Errorf("could not create security email: %v", err)
+	}
+
+	return resourceSecurityEmailRead(ctx, data, i)
+}
+
+func resourceSecurityEmailDelete(ctx context.Context, data *schema.ResourceData, i interface{}) diag.Diagnostics {
+	return diag.Errorf("deleting security email is not supported")
```
Comment (adamconnelly) [inline spacelift/resource_security_email.go:86]: This doesn't seem a great UX. Wouldn't it mean that if you had added this resource, you'd need to remove it from the Terraform state in order to be able to delete the resource?

I wonder if we should do something like this:

1. Update the API to allow the security email to be set to an empty string / null; or
2. Turn this into a no-op (maybe with a warning).

File: spacelift/resource_security_email.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+)
+
+func resourceSecurityEmail() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_security_email` represents a security email address that " +
```
Comment (adamconnelly) [inline spacelift/resource_security_email.go:16]:
```suggestion
			"`spacelift_security_email` represents an email address that " +
```

Just a nit - I don't think we need `security` in here twice.

**Review by adamconnelly — COMMENTED** [review]
I'm not convinced about the delete behaviour, but other than that it makes sense.

Can you add a test for the resource please?

File: spacelift/resource_security_email.go (hunk 1 above)
Comment (mwasilew2) [reply spacelift/resource_security_email.go:86]: > you'd need to remove it from the Terraform state in order to be able to delete the resource?

yes

I'll turn it into a warning and remove the resource from the state

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**